src/components/heading/heading.twig

```diff
@@ -20,15 +20,21 @@
   Build the class name based on the level.
 #}
 
-{% set _level_class = 'c-heading' %}
+{% set _heading_class = 'c-heading' %}
+{% set _heading_class_prefix = 'c-heading' %}
 
 {% if level is not empty %}
   {% set _level_modifier = ('level-' ~ level)|replace({ '--': '-n' }) %}
-  {% set _level_class = _level_class ~ ' ' ~ _level_class ~ '--' ~ _level_modifier %}
+  {% set _heading_class = _heading_class ~ ' ' ~ _heading_class_prefix ~ '--' ~ _level_modifier %}
+{% endif %}
+
+{% if weight is not empty %}
+  {% set _weight_modifier = 'c-heading--light' %}
+  {% set _heading_class = _heading_class_prefix ~ '--' ~ weight %}
```

Comment:
I believe you've made the requested change to the wrong line. Now `c-heading--light` is the only class on the element.

<img width="444" alt="Screen Shot 2021-07-13 at 1 44 08 PM" src="https://user-images.githubusercontent.com/69633/125522206-8fac42b1-41bc-4e48-a50d-16a214997c2f.png">